presto-main/src/main/java/com/facebook/presto/sql/planner/StatisticsAggregationPlanner.java

```diff
@@ -110,26 +110,49 @@ public TableStatisticAggregation createStatisticsAggregation(TableStatisticsMeta
         return new TableStatisticAggregation(aggregation, descriptor.build());
     }
 
-    private ColumnStatisticsAggregation createColumnAggregation(ColumnStatisticType statisticType, VariableReferenceExpression input)
+    private ColumnStatisticsAggregation createColumnAggregation(ColumnStatisticType statisticType, VariableReferenceExpression input, boolean afterTranslateExpressions)
     {
-        SymbolReference symbolReference = new SymbolReference(input.getName());
-        switch (statisticType) {
-            case MIN_VALUE:
-                return createAggregation("min", symbolReference, input.getType(), input.getType());
-            case MAX_VALUE:
-                return createAggregation("max", symbolReference, input.getType(), input.getType());
-            case NUMBER_OF_DISTINCT_VALUES:
-                return createAggregation("approx_distinct", symbolReference, input.getType(), BIGINT);
-            case NUMBER_OF_NON_NULL_VALUES:
-                return createAggregation("count", symbolReference, input.getType(), BIGINT);
-            case NUMBER_OF_TRUE_VALUES:
-                return createAggregation("count_if", symbolReference, BOOLEAN, BIGINT);
-            case TOTAL_SIZE_IN_BYTES:
-                return createAggregation(SumDataSizeForStats.NAME, symbolReference, input.getType(), BIGINT);
-            case MAX_VALUE_SIZE_IN_BYTES:
-                return createAggregation(MaxDataSizeForStats.NAME, symbolReference, input.getType(), BIGINT);
-            default:
-                throw new IllegalArgumentException("Unsupported statistic type: " + statisticType);
+        if (!afterTranslateExpressions) {
+            SymbolReference symbolReference = new SymbolReference(input.getName());
```

Comment:
I tried to use the `VariableReferenceExpression` directly, and it seems like there are test failures.